File: week 2 - arrays/readability/readability.c
#include <cs50.h>
#include <ctype.h>
#include <math.h>
#include <stdio.h>
#include <string.h>

int main(void)
{
    // Get input text from user
    string text = get_string("Text: ");

    int letters = 0;
    int sentences = 0;
    int words = 1;

    for (int i = 0, n = strlen(text); i < n; i++)
    {
        if (islower(text[i]) || isupper(text[i]))
        {
            letters++;
        }
        if (text[i] == '.' || text[i] == '!' || text[i] == '?')
        {
            sentences++;
        }
        if (text[i] == ' ')
        {
            words++;
        }
    }
    float l = letters / (float) words * 100;
    float s = sentences / (float) words * 100;

    int index = round(0.0588 * l - 0.296 * s - 15.8);

    if (index < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (index >= 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Grade %i\n", index);
    }
}

File: week 6 - python/sentimental-readability/readability.py
import cs50

text = cs50.get_string("Text: ")

letters = 0
sentences = 0
words = 1

for i in range(len(text)):
    if text[i].islower() or text[i].isupper():
        letters += 1
    if text[i] == "." or text[i] == "!" or text[i] == "?":
        sentences += 1
    if text[i] == " ":
        words += 1

l = letters / float(words) * 100
s = sentences / float(words) * 100
grade = round(0.0588 * l - 0.296 * s - 15.8)

if grade < 1:
    print("Before Grade 1")
elif grade >= 16:
    print("Grade 16+")
else:
    print("Grade " + str(grade))
